src/tests/test_magic.py

```diff
@@ -1305,3 +1305,194 @@ def test_interact_and_missing_ipywidgets_installed(ip):
+)
+def test_valid_loading_toml(tmp_empty, ip, capsys, file_content, expect, revert):
+    Path("pyproject.toml").write_text(file_content)
+    current = os.getcwd()
+    os.makedirs("sub")
+    os.chdir("sub")
+
+    ip.run_cell("%load_ext sql").result
+    out, _ = capsys.readouterr()
+
+    expect = expect % (current)
+    assert out.strip() == expect.strip()
+
+    sql = ip.find_cell_magic("sql").__self__
+    [setattr(sql, config, value) for config, value in revert.items()]
+
+
+def test_no_toml(tmp_empty, ip, capsys):
+    os.makedirs("sub")
+    os.chdir("sub")
+
+    ip.run_cell("%load_ext sql").result
+    out, _ = capsys.readouterr()
+
+    assert out == ""
+
+
+@pytest.mark.parametrize(
+    "file_content, error_msg",
+    [
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = true
+autocommit = true
+""",
+            "Duplicate key found : 'autocommit'",
+        ),
+        (
+            """
+[tool.jupySql.SqlMagic]
+autocommit = true
+""",
+            "'jupySql' is an invalid section name. Did you mean 'jupysql'?",
+        ),
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = True
+""",
+            (
+                "Invalid value 'True' in 'autocommit = True'. "
+                "Valid boolean values: true, false"
+            ),
+        ),
+        (
+            """
+[tool.jupysql.SqlMagic]
+autocommit = invalid
+""",
+            (
+                "Invalid value 'invalid' in 'autocommit = invalid'. "
+                "To use str value, enclose it with ' or \"."
+            ),
+        ),
+    ],
+)
+def test_error_on_toml_parsing(tmp_empty, ip, capsys, file_content, error_msg):
+    Path("pyproject.toml").write_text(file_content)
+    toml_dir = os.getcwd()
+    found_statement = "Found pyproject.toml from '%s'" % (toml_dir)
+    os.makedirs("sub")
+    os.chdir("sub")
+
+    exec_result = ip.run_cell("%load_ext sql")
+    out, _ = capsys.readouterr()
+
+    assert out.strip() == found_statement
+    assert exec_result.error_in_exec.error_type == "ConfigurationError"
+    assert str(exec_result.error_in_exec) == error_msg
+
+
+def test_valid_and_invalid_configs(tmp_empty, ip, capsys):
+    Path("pyproject.toml").write_text(
+        """
+[tool.jupysql.SqlMagic]
+autocomm = true
+autop = false
+autolimit = "text"
+invalid = false
+displaycon = false
+"""
+    )
+    toml_dir = os.getcwd()
+    os.makedirs("sub")
+    os.chdir("sub")
+
+    ip.run_cell("%load_ext sql")
+    out, _ = capsys.readouterr()
+    expect = [
+        "Found pyproject.toml from '%s'" % (toml_dir),
+        "'autocomm' is an invalid configuration. Did you mean 'autocommit'?",
+        (
+            "'autop' is an invalid configuration. "
+            "Did you mean 'autopandas', or 'autopolars'?"
+        ),
+        (
+            "'text' is an invalid value for 'autolimit'. "
+            "Please use int value instead."
+        ),
+        """
+Settings changed:
+
++------------+-------+
+|   Config   | value |
```

Comment:
can we extract this table in code and check the columns instead of hardcoding the table structure?

Reply:
Extracted rows instead and check them in the assert statement below using regular expression.